spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -114,11 +117,22 @@
   private final Long snapshotId;
   private final boolean refreshEagerly;
   private final Set<TableCapability> capabilities;
+  private String branch;
   private StructType lazyTableSchema = null;
   private SparkSession lazySpark = null;
 
   public SparkTable(Table icebergTable, boolean refreshEagerly) {
-    this(icebergTable, null, refreshEagerly);
+    this(icebergTable, (Long) null, refreshEagerly);
+  }
+
+  public SparkTable(Table icebergTable, String branch, boolean refreshEagerly) {
+    this(icebergTable, refreshEagerly);
+    this.branch = branch;
+    ValidationException.check(
+        branch != null
```

Comment:
Isn't it okay for `branch` to be null? i know we would prefer using the `SparkTable(Table,Long,boolean)` constructor, but I think if branch is set to null in this constructor then everything works the same.

This check could be `ValidationException.check(branch == null || MAIN_BRANCH.equals(branch) || table.snapshot(branch) != null, ...)` which would be simpler to read.

Reply:
Makes sense, updated